src/tools/perf/perftest.c

```diff
@@ -711,13 +735,15 @@ static ucs_status_t setup_mpi_rte(struct perftest_context *ctx)
     ucs_trace_func("");
 
     MPI_Comm_size(MPI_COMM_WORLD, &size);
-    if (size != 2) {
+    if (size != 2 &&
+            !(UCX_PERF_TEST_FLAG_LOOPBACK_CONN & ctx->params.super.flags)) {
         ucs_error("This test should run with exactly 2 processes (actual: %d)", size);
         return UCS_ERR_INVALID_PARAM;
     }
 
     MPI_Comm_rank(MPI_COMM_WORLD, &rank);
-    if (rank == 1) {
+    if (rank == 1 ||
+            UCX_PERF_TEST_FLAG_LOOPBACK_CONN & ctx->params.super.flags) {
         ctx->flags |= TEST_FLAG_PRINT_RESULTS;
```

Comment:
In `setup_sock_rte`, if we're in loopback, `UCX_PERF_TEST_FLAG_LOOPBACK_CONN` is also added. Need to do it here as well?

Reply:
I guess you meant TEST_FLAG_PRINT_TEST... I don't add it here because the original code doesn't add this flag as well...